### DISCUSSION TITLE
Implement prompt caching for Claude adapter.

### DISCUSSION THREAD

**Discussion by cperion** [olimorris/codecompanion.nvim#111] [Ideas]
First of all, thank you for this great tool ! I have been using aider form Paul Gautier and I always longed for something that would integrate nicely with neovim. Thank You !

Now : https://www.anthropic.com/news/prompt-caching
Implementing prompt caching would allow for better performance and less costly api calls for Claude models.
It should not be very hard as the caching API is pretty simple, however I think we should be careful about the prompt "order".

**Comment by olimorris** [comment]
Thanks for sharing. I note this was also raised in #99.

I'm currently refactoring the way we work with messages in the chat buffer in order to open up this kind of functionality. Technically, the `form_messages` handler on the adapter could be used to enable this but it will be a bit clunky.

Will be amazing to cache some of the large requests like buffer variables.

**Reply by cperion** [reply]
Sorry for the duplicate. Glad you are working on it ! Bon courage !

**Reply by olimorris** [reply]
Haha no worries. Even I couldn't find the issue that originally mentioned it 😆

**Reply by olimorris** [reply]
Please see #126 . Would love your feedback.
